### PR TITLE
Allow mediaSelect to upload media files using dragable upload media control

### PR DESCRIPTION
**Pull request by masmerino13**

### CODE REVIEW

File: src/apps/common/submodules/mediaSelect/mediaSelect.js [hunk 1]
```diff
@@ -230,10 +242,31 @@ define(function(require) {
 			});
 
 			template.find('.upload-toggle').on('click', function() {
-				if ($(this).hasClass('active')) {
-					template.find('.upload-div').stop(true, true).slideUp();
+				if (args.dragableUpload) {
```
[the remark this answers is not in the document]
Comment (masmerino13) [reply src/apps/common/submodules/mediaSelect/mediaSelect.js]: I'm not sure about it, how this should works?

This new prop is just to handle the upload media link and it's visible (the upload media link) only in default template.

I'm missing something?

File: src/apps/common/submodules/mediaSelect/mediaSelect.js [hunk 2]
```diff
@@ -11,6 +11,12 @@ define(function(require) {
 			'common.mediaSelect.render': 'mediaSelectRender'
 		},
 
+		appFlags: {
+			mediaSelect: {
+				allowedFiles: ['audio/mp3', 'video/mp4']
```
[the remark this answers is not in the document]
Comment (masmerino13) [reply src/apps/common/submodules/mediaSelect/mediaSelect.js]: Those will be the default values and I will add a new optinal prop that should allow to indicate the `allowed file` formats

[the remark this answers is not in the document]
Comment (masmerino13) [reply src/apps/common/submodules/mediaSelect/mediaSelect.js]: any ideas are welcome :)

File: src/apps/common/submodules/mediaSelect/mediaSelect.js [hunk 3]
```diff
@@ -209,8 +215,14 @@ define(function(require) {
 			});
 		},
 
-		mediaSelectBindDefaultTemplate: function(template) {
+		mediaSelectBindDefaultTemplate: function(template, args) {
```
[the remark this answers is not in the document]
Comment (masmerino13) [reply src/apps/common/submodules/mediaSelect/mediaSelect.js]: What's about add those comments in the main render method?

File: src/apps/common/submodules/mediaSelect/mediaSelect.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (azefiel) [reply src/apps/common/submodules/mediaSelect/mediaSelect.js]: I don't think there should a default for this: audio and file formats are equally used throughout the UI and each file upload component should define that format.

It definitely should be configurable through an mandatory properties though.

File: src/apps/common/submodules/mediaSelect/mediaSelect.js [hunk 4]
```diff
@@ -11,6 +11,22 @@ define(function(require) {
...
+		},
+
+		/**
+		 * @param {Object} args
+		 * @param {jQuery} args.container
+		 * @param {String} [args.skin]
+		 * @param {Function} args.callback
+		 * @param {String} [args.label]
+		 * @param {Array} args.options
+		 * @param {Array} [args.allowedFiles]
```
Comment (azefiel) [inline src/apps/common/submodules/mediaSelect/mediaSelect.js]: [MIME](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types) is commonly used to refer to this sort of types:
```suggestion
		 * @param {Array} args.mimeTypes
```

File: src/apps/common/submodules/mediaSelect/mediaSelect.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (azefiel) [reply src/apps/common/submodules/mediaSelect/mediaSelect.js]: My bad, seems like I confused this common control with the `mediaSelector` one...

File: src/apps/common/submodules/mediaSelect/mediaSelect.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (masmerino13) [reply src/apps/common/submodules/mediaSelect/mediaSelect.js]: sounds good to me

**Pull request merged**